src/credentials/GroupcastDataProvider.h

```diff
@@ -0,0 +1,96 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/basic-types.h>
+#include <credentials/FabricTable.h>
+#include <crypto/CHIPCryptoPAL.h>
+#include <crypto/SessionKeystore.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/support/CommonIterator.h>
+#include <lib/support/Pool.h>
+#include <lib/support/Span.h>
+#include <string.h>
+
+namespace chip {
+namespace Groupcast {
+
+constexpr size_t kEndpointsMax = 20;
+constexpr size_t kIteratorMax  = CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS;
+
+struct Group
```

Comment:
Is there a way in which we could leverage `src/app/storage/TableEntry.h` for the group storage? As it is being used in FabricTableImpl and SceneTableImpl in a similar way.